internal/middleware/etag_middleware.go

```diff
@@ -0,0 +1,67 @@
+package middleware
+
+import (
+	"errors"
+	"fmt"
+	"hash/crc64"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/RHEnVision/provisioning-backend/internal/ctxval"
+)
+
+var InvalidETagErr = errors.New("empty etag provided")
+
+type ETag struct {
+	Name       string
+	Expiration time.Duration
+	Value      string
+}
+
+type ETagValueFunc func() ETag
```

Comment:
This is the function which returns the type above which contains all information for etag to be generated from.